custom_components/wyzeapi/binary_sensor.py

```diff
@@ -222,3 +245,91 @@ def process_update(self, camera: Camera) -> None:
 
         self.schedule_update_ha_state()
+
+
+class WyzeIrrigationZoneRunning(BinarySensorEntity):
+    """Representation of a Wyze Irrigation Zone Running binary sensor."""
+
+    _attr_device_class = BinarySensorDeviceClass.RUNNING
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, irrigation_service: IrrigationService, irrigation: Irrigation, zone_number: int, zone_name: str) -> None:
+        """Initialize the irrigation zone running sensor."""
+        self._irrigation_service = irrigation_service
+        self._device = irrigation
+        self._zone_number = zone_number
+        self._zone_name = zone_name
+        self._running = False
+
+    @property
+    def name(self) -> str:
+        """Return the name of the zone."""
+        return f"{self._zone_name}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID for the zone."""
+        return f"Running {self._device.mac}-zone-{self._zone_number}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the zone is running."""
+        return self._running
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device.mac)},
+            name=self._device.nickname,
+            manufacturer="WyzeLabs",
+            model=self._device.product_model,
+            serial_number=self._device.sn,
+            connections={(dr.CONNECTION_NETWORK_MAC, self._device.mac)},
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return extra state attributes."""
+        return {
+            "zone_number": self._zone_number,
+            "zone_name": self._zone_name,
+        }
+
+    async def async_update(self) -> None:
+        """Update the sensor state."""
+        try:
+            schedule_data = await self._irrigation_service.get_schedule_runs(self._device)
+            # Check if this specific zone is running
+            if schedule_data.get("running", False):
+                running_zone_number = schedule_data.get("zone_number")
+                self._running = running_zone_number == self._zone_number
+            else:
+                self._running = False
+        except Exception as e:
+            _LOGGER.error("Failed to update zone running status for device %s zone %s: %s", 
+                         self._device.mac, self._zone_number, str(e))
+            self._running = False
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to updates and set up periodic updates."""
+        # Set up periodic updates every minute for zone running status
+        self._unsub_periodic = async_track_time_change(
+            self.hass, 
+            self._async_periodic_update, 
+            second=0  # Update every minute at second 0
+        )
+        return await super().async_added_to_hass()
```

Comment:
This is probably more of a question for @SecKatie, but I'm thinking instead of adding a specific time interval sensor, shouldn't we be adding this to the update manager like we do for all the other platforms? For continuity and that would keep the total update interval in check if lots of devices are being updated.

Reply:
Really we should be moving to [the UpdateCoordinator pattern](https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities). For this change I am fine with this as is unless @FeatherKing wants to take a stab at building an UpdateCoordinator for us. My only other recommendation would be for us to make it a longer delay. Something like 2-5 minutes should be better.